pkg/apis/operator/v1alpha1/tektonchain_types.go

```diff
@@ -62,8 +62,13 @@ type TektonChainSpec struct {
 	Config Config `json:"config,omitempty"`
 }
 
-// Chain defines the field to provide chain configuration
 type Chain struct {
+	ChainProperties `json:",inline"`
+	ControllerEnvs  []ControllerEnv `json:"controllerEnvs,omitempty"`
```

Comment:
IMHO, we can use `[]corev1.EnvVar` in the place of `[]ControllerEnv`,
With this we can supply the value from secret too. also supports plan `key`, `value` approach.
[EnvVar Doc](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#envvar-v1-core)
```
impoert "corev1 "k8s.io/api/core/v1"

ControllerEnvs []corev1.EnvVar `json:"controllerEnvs,omitempty"`

```

an example to pass a secret via `controllerEnvs`,
```
controllerEnvs:
  - name: MONGO_SERVER_URL
    valueFrom:
      secretKeyRef:
        name: custom_secret_name
        key: mongo_server_url
```

Reply:
Yeah makes sense, but not I'm confused, what to add in the docs

Reply:
yup we should take the struct from containers, it will be beneficial to not hardcode in CR